File: leetcode/217. Contains Duplicate/Solution.kt
/**
 * Created by Inno Fang on 2018/2/9.
 */


/**
 * 18 / 18 test cases passed.
 * Status: Accepted18 / 18 test cases passed.
 * Runtime: 496 ms
 */
class Solution {
    fun containsDuplicate(nums: IntArray): Boolean {
        return nums.toHashSet().size < nums.size
    }
}

File: LeetCode/217. Contains Duplicate/Solution.cpp
#include <unordered_map>
using namespace std;

/**
 * Status: Accepted
 * Runtime: 32 ms
 */
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> dict;
        for (int i = 0; i < nums.size(); ++i) {
            if (dict.find(nums[i]) != dict.end()) dict[nums[i]] ++;
            else dict[nums[i]] = 1;
        }
        return nums.size() != dict.size();
    }
};